traffic_ops/testing/api/v4/servers_test.go

```diff
@@ -423,6 +400,31 @@ func TestServers(t *testing.T) {
 	})
 }
 
+func setupCacheGroupWithASN(t *testing.T) {
+	// Add a new asn for one of the cachegroups
+	cgResp, _, err := TOSession.GetCacheGroups(client.RequestOptions{QueryParameters: url.Values{"name": {"topology-mid-cg-01"}}})
+	if err != nil {
+		t.Fatalf("couldn't get cachegroups: %v", err)
+	}
+	if len(cgResp.Response) != 1 {
+		t.Fatalf("expected 1 cachegroup, but got %d", len(cgResp.Response))
+	}
+	if cgResp.Response[0].ID == nil {
+		t.Fatalf("ID of cachegroup is nil")
+	}
+
+	asn := tc.ASN{
+		ASN:          1111,
+		Cachegroup:   "topology-mid-cg-01",
+		CachegroupID: *cgResp.Response[0].ID,
+	}
+
+	_, _, err = TOSession.CreateASN(asn, client.NewRequestOptions())
+	if err != nil {
+		t.Fatalf("couldn't create ASN: %v", err)
+	}
+}
+
```

Comment:
This can just be added to the asns array in tc-fixtures since its a prerequisite. No need for a setup function. 
e.g:
```
        {
            "asn": 1111,
            "cachegroupName": "topology-mid-cg-01"
        }
```

Reply:
So I think there is a bug in the way our tests are reading the ASNs. If you add that section into the fixtures file, and do a `TOSession.GetASNS()`, the results that come back are:
`{1111 originCachegroup 18999 145 2022-08-22 17:48:28-06}
{8888 originCachegroup 18999 143 2022-08-22 17:48:28-06}
{9999 originCachegroup 18999 144 2022-08-22 17:48:28-06}`

This is incorrect. I'll create a separate issue for that.

Reply:
Oh I see it. If you can assign it to me I'll fix it.

Reply:
@ericholguin here you go: https://github.com/apache/trafficcontrol/issues/7033